test/e2e/scheduling/predicates.go

```diff
@@ -192,6 +194,135 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		verifyResult(cs, podsNeededForSaturation, 1, ns)
 	})
 
+	// This test verifies we don't allow scheduling of pods in a way that sum of limits +
+	// associated overhead is greater than machine's capacity.
+	// It assumes that cluster add-on pods stay stable and cannot be run in parallel
+	// with any other test that touches Nodes or Pods.
+	// Because of this we need to have precise control on what's running in the cluster.
+	// Test scenario:
+	// 1. Find the amount CPU resources on each node.
+	// 2. Create one pod with affinity to each node that uses 70% of the node CPU.
+	// 3. Wait for the pods to be scheduled.
+	// 4. Create another pod with no affinity to any node that needs 20% of the largest node CPU for the workload and
+	//    an overhead set as 30% of the largest node CPU.
+	// 5. Make sure this additional pod is not scheduled.
+
+	ginkgo.It("validates pod overhead is considered along with resource limits of pods that are allowed to run", func() {
+		WaitForStableCluster(cs, masterNodes)
+		nodeMaxAllocatable := int64(0)
+		nodeToAllocatableMap := make(map[string]int64)
+		for _, node := range nodeList.Items {
+			nodeReady := false
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+					nodeReady = true
+					break
+				}
+			}
+			if !nodeReady {
+				continue
```

Comment:
it's not clear to me what happens with this node skip and when it can happen.
when can a Node not be ready during this test runtime?

should there be a least a log message about it?

Reply:
@bsalamat had this in place in the original test that this is based on.  AFAIU this was done to add some robustness to the test, which was flakey otherwise (ie, let's only apply the pods to nodes which are actually available). I'm not sure what value the message would bring.

Bobby, can you help provide input here, and PTAL @ the PR in general?

Reply:
Since we are dealing with a real cluster in e2e tests, there is a chance a node may not become ready for various reasons (network issues, driver issues, etc). This check was there to avoid flakes in clusters with unready nodes.

Delegating to @ahg-g to provide a more thorough review.